### PR TITLE
fix(docker):Update couchdb3.1 ubuntu20.04 liferay7.3.4 postgresql12

### PR DESCRIPTION
**Pull request by KoukiHama**
Signed-off-by: Kouki Hama <kouki1.hama@toshiba.co.jp>

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: #1322  #1331

Build docker image by using this commad

> docker build -t sw360 .

Run docker 

> docker run -it -p 8090:8080 -p 5985:5984 -p 5435:5432 sw360

and set Liferay7.3.4 GUI 
 

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by KoukiHama** [comment]
I have finished to update almost all, but I was not able to solve these Error


> 2021-07-27 00:43:22 INFO  Sw360CustomHeaderAuthenticationFilter:105 - AuthenticationFilter is NOT active!
2021-07-27 00:43:22 WARN  AnnotationConfigEmbeddedWebApplicationContext:551 - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'OAuthClientRepository' defined in file [/app/liferay-ce-portal-7.3.4-ga5/tomcat-9.0.33/webapps/authorization/WEB-INF/classes/org/eclipse/sw360/rest/authserver/client/persistence/OAuthClientRepository.class]: Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.eclipse.sw360.rest.authserver.client.persistence.OAuthClientRepository]: Constructor threw exception; nested exception is org.ektorp.DbAccessException: 401:Unauthorized
URI: /sw360oauthclients/
Response Body: 
{
  "error" : "unauthorized",
  "reason" : "You are not a server admin."
}
2021-07-27 00:43:22 ERROR SpringApplication:771 - Application startup failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'OAuthClientRepository' defined in file [/app/liferay-ce-portal-7.3.4-ga5/tomcat-9.0.33/webapps/authorization/WEB-INF/classes/org/eclipse/sw360/rest/authserver/client/persistence/OAuthClientRepository.class]: Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.eclipse.sw360.rest.authserver.client.persistence.OAuthClientRepository]: Constructor threw exception; nested exception is org.ektorp.DbAccessException: 401:Unauthorized
URI: /sw360oauthclients/
Response Body: 
{
  "error" : "unauthorized",
  "reason" : "You are not a server admin."
}

**Comment by JaideepPalit** [comment]
Hi @KoukiHama . I guess you need to place `application.yml` in `/etc/sw360/authorization/`, with correct username and pwd

**Comment by KoukiHama** [comment]
Thank you @JaideepPalit .

I was able to remove ouath client error by adding application.yml in /etc/sw360/authorization/ !

But, other problem are happened ...

Component and Project pages are not visible.  (I have already set user groups. )

image
 
In the container CouchDB3.1 works fine.

>  curl http://admin:password@127.0.0.1:5984
{"couchdb":"Welcome","version":"3.1.1","git_sha":"ce596c65d","uuid":"d3f19f9e1babc3dc92b3bd58d9f00ca4","features":["access-ready","partitioned","pluggable-storage-engines","reshard","scheduler"],"vendor":{"name":"The Apache Software Foundation"}}

Tomcat.log  implies that some Credentials are incorrect.  But I set user and password  coucdb.properties in /etc/sw360
**Should I set couchdb's user and password in other files aside from coucdb.propeties and  application.yml?**

> couchdb.url = http://localhost:5984
couchdb.user = admin
couchdb.password = password
couchdb.database = sw360db
couchdb.usersdb = sw360users
couchdb.attachments = sw360attachments
lucenesearch.limit = 1000
 
>Caused by: com.cloudant.client.org.lightcouch.CouchDbException: 401 Credentials are incorrect for server http://localhost:5984/_session.
        at com.cloudant.client.org.lightcouch.CouchDbClient.execute(CouchDbClient.java:567) ~[cloudant-client-2.19.1.jar:?]
        at com.cloudant.client.org.lightcouch.CouchDbClient.executeToInputStream(CouchDbClient.java:648) ~[cloudant-client-2.19.1.jar:?]
        at com.cloudant.client.org.lightcouch.CouchDbClient.get(CouchDbClient.java:389) ~[cloudant-client-2.19.1.jar:?]
        at com.cloudant.client.org.lightcouch.CouchDbClient.getAllDbs(CouchDbClient.java:243) ~[cloudant-client-2.19.1.jar:?]
        at com.cloudant.client.api.CloudantClient.getAllDbs(CloudantClient.java:292) ~[cloudant-client-2.19.1.jar:?]
        at org.eclipse.sw360.datahandler.cloudantclient.DatabaseInstanceCloudant.checkIfDbExists(DatabaseInstanceCloudant.java:37) ~[datahandler-13.4.0-SNAPSHOT.jar:?]
        at org.eclipse.sw360.datahandler.cloudantclient.DatabaseInstanceCloudant.createDB(DatabaseInstanceCloudant.java:33) ~[datahandler-13.4.0-SNAPSHOT.jar:?]
        at org.eclipse.sw360.datahandler.cloudantclient.DatabaseConnectorCloudant.<init>(DatabaseConnectorCloudant.java:64) ~[datahandler-13.4.0-SNAPSHOT.jar:?]
        at org.eclipse.sw360.fossology.config.FossologyConfig.configContainerRepository(FossologyConfig.java:41) ~[src-fossology-13.4.0-SNAPSHOT.jar:?]
        at org.eclipse.sw360.fossology.config.FossologyConfig$$EnhancerBySpringCGLIB$$e2210a9a.CGLIB$configContainerRepository$3(<generated>) ~[src-fossology-13.4.0-SNAPSHOT.jar:?]
        at org.eclipse.sw360.fossology.config.FossologyConfig$$EnhancerBySpringCGLIB$$e2210a9a$$FastClassBySpringCGLIB$$c6b078a4.invoke(<generated>) ~[src-fossology-13.4.0-SNAPSHOT.jar:?]
        at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244) ~[spring-core-5.2.7.RELEASE.jar:5.2.7.RELEASE]
        at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:331) ~[spring-context-5.2.7.RELEASE.jar:5.2.7.RELEASE]
        at org.eclipse.sw360.fossology.config.FossologyConfig$$EnhancerBySpringCGLIB$$e2210a9a.configContainerRepository(<generated>) ~[src-fossology-13.4.0-SNAPSHOT.jar:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154) ~[spring-beans-5.2.7.RELEASE.jar:5.2.7.RELEASE]
        at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:650) ~[spring-beans-5.2.7.RELEASE.jar:5.2.7.RELEASE]
        at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:483) ~[spring-beans-5.2.7.RELEASE.jar:5.2.7.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1338) ~[spring-beans-5.2.7.RELEASE.jar:5.2.7.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1177) ~[spring-bea:

**Comment by JaideepPalit** [comment]
Not sure , Could you try removing extra space from username and pwd
Since it is saying - 401 Credentials are incorrect for server

image

**Comment by KoukiHama** [comment]
@JaideepPalit  Thank you! Issues was solved by your advice!

image

image

**Review by mcjaeger — COMMENTED** [review]
I tested it with current master (import spdx BOM, add some SPDX files to created releases and generated OSS license info) -> all good. 

Search does not work for me however, but ...

**Comment by mcjaeger** [comment]
BTW. about the search, it is 
```
2021-07-27 20:03:13,703 WARN [DefaultHttpClient] Authentication error: Unable to respond to any of these challenges: {}
2021-07-27 20:03:13 ERROR LuceneAwareDatabaseConnector:163 - Retried with stale parameter.Getting error with reponse code=500
```

File: scripts/docker-config/install_scripts/install_init_postgres_script.sh [hunk 1]
```diff
@@ -26,14 +26,18 @@ init_db_and_user() {
...
   /etc/init.d/postgresql restart
 }
 
 install_postgres() {
-  DEBIAN_FRONTEND=noninteractive apt-get install postgresql-10 -y --no-install-recommends
+  apt-get install curl -y --no-install-recommends
+  curl -L https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
+  echo "deb http://apt.postgresql.org/pub/repos/apt focal-pgdg main" | tee -a /etc/apt/sources.list
+  apt-get update
+  DEBIAN_FRONTEND=noninteractive apt-get install postgresql-11 -y --no-install-recommends
```
[the remark this answers is not in the document]
Comment (KoukiHama) [reply scripts/docker-config/install_scripts/install_init_postgres_script.sh:40]: OK I will change postgressql version 11 to 12.

**Comment by KoukiHama** [comment]
For searching function and couchdb-lucene, I plan to add this setting like CouchDB version2.x

> [httpd_global_handlers]
> _fti = {couch_httpd_proxy, handle_proxy_req, <<"http://127.0.0.1:8080/couchdb-lucene">>}

But I don't know, this setting works well or not, even if   [httpd_global_handlers] is not exists in default,ini
https://github.com/apache/couchdb/blob/main/rel/overlay/etc/default.ini

**Comment by KoukiHama** [comment]
For GitHub action I updated and cleaned commit logs.

**Comment by JaideepPalit** [comment]
For search, I guess user name , pwd needs to be set in `couchdb-lucene.ini` with
https://github.com/eclipse/sw360/blob/c7b1335d4561c1736285d37baa61277c6420ae02/scripts/docker-config/install_scripts/build_couchdb_lucene.sh#L19
```
# Allow leading wildcard?
allowLeadingWildcard=true

# couchdb server mappings

[local]
url = http://username:pwd@localhost:5984/
```

**Comment by KoukiHama** [comment]
@JaideepPalit  Your guess is right!

By setting User and Password in couchdb-lucene.ini 
Searching wroks fine!

New one (with http://username:pwd@localhost:5984/ )
image

Old one (without http://username:pwd@localhost:5984/)
image

**Comment by mcjaeger** [comment]
search works for me as well:

Screenshot 2021-08-18 at 14 26 43

installation (build and run and LR setup)  does not show issues

**Pull request merged**